File: python/divisible_and_non_divisible_sums_difference_2894.py
class Solution:
    def differenceOfSums(self, n: int, m: int) -> int:
        num1 = sum(x for x in range(1, n + 1) if x % m != 0)
        num2 = sum(x for x in range(1, n + 1) if x % m == 0)
        return num1 - num2

File: rust/src/divisible_and_non_divisible_sums_difference_2894.rs
use crate::Solution;
impl Solution {
    pub fn difference_of_sums(n: i32, m: i32) -> i32 {
        let (mut num1, mut num2) = (0, 0);

        for x in 1..=n {
            if x % m == 0 {
                num2 += x; // Sum of numbers divisible by m
            } else {
                num1 += x; // Sum of numbers not divisible by m
            }
        }
        num1 - num2
    }
}
